rust/lance/src/dataset.rs

```diff
@@ -1466,6 +1436,56 @@ impl Dataset {
     }
 }
 
+impl Dataset {
+    /// Remove columns from the dataset.
+    ///
+    /// This is a metadata-only operation and does not remove the data from the
+    /// underlying storage. In order to remove the data, you must subsequently
+    /// call `compact_files` to rewrite the data without the removed columns and
+    /// then call `cleanup_files` to remove the old files.
+    pub async fn drop_columns(&mut self, columns: &[&str]) -> Result<()> {
+        // Check if columns are present in the dataset and construct the new schema.
+        for col in columns {
+            if self.schema().field(col).is_none() {
+                return Err(Error::invalid_input(
+                    format!("Column {} does not exist in the dataset", col),
+                    location!(),
+                ));
+            }
+        }
+
+        let columns_to_remove = self.manifest.schema.project(columns)?;
```

Comment:
I think project can handle nested column references right?  What happens if a nested reference is provided?

Reply:
It will drop the nested column. There is a unit test for this: https://github.com/lancedb/lance/blob/812d9f3f119000a29fad2e74ab38c8d164d45b69/rust/lance/src/dataset.rs#L2706